soroban-sdk/src/storage.rs

```diff
@@ -165,6 +167,46 @@ impl Storage {
         }
     }
 
+    // TODO: Should we provide these methods, or should we just provide 
```

Comment:
I'm not sure the approach here where the metadata key is obfuscated away is safe or a good idea. Instead, we could remove the storage::set_metadata/get_metadata methods and just give the user the option to use the key. @dmkozh thoughts?

Reply:
I would probably lean towards smaller/simpler interface for storage and define the key in the utility library or maybe define these functions in the utility library (that's what I've decided to do with the test helper for calling check_auth). I'm not feeling like we're providing enough value here in exchange of complicating the interface/making the maintenance harder. 

But I'd also be curious to see @leighmcculloch opinion as well.

Reply:
Yeah I don't think these storage methods are necessary.